fastlane/lib/fastlane/actions/set_info_plist_value.rb

```diff
@@ -45,16 +105,32 @@ def self.available_options
           FastlaneCore::ConfigItem.new(key: :key,
                                        env_name: "FL_SET_INFO_PLIST_PARAM_NAME",
                                        description: "Name of key in plist",
-                                       optional: false),
+                                       conflicting_options: [:map],
+                                       optional: true),
           FastlaneCore::ConfigItem.new(key: :subkey,
                                        env_name: "FL_SET_INFO_PLIST_SUBPARAM_NAME",
                                        description: "Name of subkey in plist",
+                                       conflicting_options: [:map],
                                        optional: true),
           FastlaneCore::ConfigItem.new(key: :value,
                                        env_name: "FL_SET_INFO_PLIST_PARAM_VALUE",
                                        description: "Value to setup",
                                        is_string: false,
-                                       optional: false),
+                                       conflicting_options: [:map],
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :map,
+                                       env_name: "FL_SET_INFO_PLIST_PARAM_MAP",
+                                       description: "Map of keys and values in plist to be merged",
+                                       type: Hash,
+                                       conflicting_options: [:key, :subkey, :value],
+                                       optional: true),
+          FastlaneCore::ConfigItem.new(key: :replace,
+                                       env_name: "FL_SET_INFO_PLIST_PARAM_REPLACE",
+                                       description: "Replace plist with map instead of merging",
+                                       is_string: false,
+                                       default_value: false,
+                                       conflicting_options: [:key, :subkey, :value],
+                                       optional: true),
```

Comment:
Should there be a verify block here to make sure that `:map` is passed in?

Also, sort of related, is there a better word than `map` as that is not a commonly used word in ruby nor iOS programming? Might make more sense as `hash` or `dictionary`? Curious if you have any thoughts about that?

Reply:
I been thinking the same thing :|
To me hash sounds better since we are working with ruby objects on our end.

Reply:
Could you give an example on how to write a verify_block to check another config_item. I cannot find it anywhere else in the code base...